internal/app/connectconformance/test_case_library.go

```diff
@@ -408,6 +409,20 @@ func populateExpectedUnaryResponse(testCase *conformancev1.TestCase) error {
 	case *conformancev1.UnaryResponseDefinition_Error:
 		// If an error was specified, it should be returned in the response
 		expected.Error = respType.Error
+
+		if respType.Error.Details == nil {
```

Comment:
We also want to do this in server responses where the number of response messages is zero.